compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4973,6 +4973,14 @@ object Types {
         foldOver(n + 1, tp)
       case tp: TypeRef if tp.info.isTypeAlias =>
         apply(n, tp.superType)
+      case tp: TermRef =>
+        apply(n, tp.underlying)
+      case tp: TypeParamRef =>
+        ctx.typerState.constraint.entry(tp) match {
```

Comment:
I believe this is equivalent to
```
apply(n, ctx.typeComparer.bounds(tp))
```
But that also points to a problem: What if the type is F-bounded? Would that not give an infinite recursion?